src/lib/evaluate.ts

```diff
@@ -178,8 +228,129 @@ export async function evaluatePromptOnTestSuite(params: {
     }
   }
   return {
-    success: passedTests === testCases.length,
+    success: passedTests == testCases.length,
     passedTests,
     totalTests: testCases.length,
+    totalUsage,
+  };
+}
+
+/**
+ * Compresses a given text using a model and a prompt
+ * @param params Parameters for compressing text
+ * @returns The compressed text and the token usage for the compression
+ */
+async function getCompressedTask(params: {
+  task: string;
+  compressingPrompt: string;
+  model: string;
+}): Promise<{ compressedTask: string; usage: OpenAI.CompletionUsage }> {
+  const { task, compressingPrompt, model } = params;
+  const openai = createOpenAIClient();
+  await new Promise(r => setTimeout(r, 500)); // Rate limit friendly
+  const response = await openai.chat.completions.create({
+    model,
+    messages: [
+      { role: 'system', content: compressingPrompt },
+      { role: 'user', content: task },
+    ],
+  });
+
+  const compressedTask = response.choices[0]?.message?.content;
+  if (!compressedTask) {
+    throw new Error('Failed to get compressed task from model.');
+  }
+  if (!response.usage) {
+    throw new Error('No usage data in compression response');
+  }
+
+  return { compressedTask, usage: response.usage };
+}
+
+/**
+ * Evaluates a compressing prompt by testing it against a test suite
+ * and calculating the compression ratio and impact on performance.
+ * @param params Parameters for evaluating the compression
+ * @returns A detailed result of the compression evaluation
+ */
+export async function evaluateCompression(params: {
+  testCases: TestCase[];
+  compressingPrompt: string;
+  compressionModel: string;
+  evaluationModel: string;
+  attemptsPerTest?: number;
+}): Promise<CompressionEvalResult> {
+  const {
+    testCases,
+    compressingPrompt,
+    compressionModel,
+    evaluationModel,
+    attemptsPerTest = 1,
+  } = params;
+
+  // 1. Baseline evaluation
+  const uncompressedResults = await evaluatePromptOnTestSuite({
+    testCases,
+    model: evaluationModel,
+    attemptsPerTest,
+  });
```

Comment:
We ensure these pass the tests before we insert them into the database. Modify the function that inserts them into the DB to store token sizes too.

Reply:
This step is not needed, the tests must have all the data computed once